src/components/molecules/VenueChat/VenueChat.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext, useEffect, useState, useMemo, FC } from "react";
+import { ChatContext } from "components/context/ChatContext";
+import "./CampChat.scss";
+import { useUser } from "hooks/useUser";
+import { useSelector } from "hooks/useSelector";
+import { useVenueId } from "hooks/useVenueId";
+import { getDaysAgoInSeconds } from "utils/time";
+import { VENUE_CHAT_AGE_DAYS } from "settings";
+import ChatBox from "components/molecules/Chatbox";
+import useRoles from "hooks/useRoles";
+
+interface ChatOutDataType {
+  messageToTheBand: string;
+}
+
+const VenueChat: FC = () => {
+  const venueId = useVenueId();
+  const { userRoles } = useRoles();
+  const { user } = useUser();
+  const { chats, venue } = useSelector((state) => ({
+    chats: state.firestore.ordered.venueChats,
+    venue: state.firestore.data.currentVenue,
+  }));
+
+  const [isMessageToTheBarSent, setIsMessageToTheBarSent] = useState(false);
+
+  useEffect(() => {
+    if (isMessageToTheBarSent) {
+      setTimeout(() => {
+        setIsMessageToTheBarSent(false);
+      }, 2000);
+    }
+  }, [isMessageToTheBarSent]);
+
+  const chatContext = useContext(ChatContext);
+
+  const onMessageSubmit = async (data: ChatOutDataType) => {
+    chatContext &&
+      user &&
+      chatContext.sendRoomChat(user.uid, venueId!, data.messageToTheBand);
+  };
+
+  function roundToNearestHour(seconds: number) {
+    const oneHour = 60 * 60;
+    return Math.floor(seconds / oneHour) * oneHour;
+  }
```

Comment:
Saw this function defined in at least one other place, would be good to extract to our time helpers

- https://github.com/sparkletown/sparkle/blob/staging/src/utils/time.js

Also, as a general 'rule', if a function/`const`/etc like this is doesn't close around any component state, then it's better to define them outside of the component render as a static.

Reply:
fixed in 77fc5d6d1829a42da2152d00871cde0b6f47269d with `src/utils/time.js`